### PR TITLE
bluetooth: upload bt local bug fix and features

### PR DESCRIPTION
**Pull request by chengkai15**
*Note: Please adhere to [Contributing Guidelines](https://github.com/apache/nuttx/blob/master/CONTRIBUTING.md).*

## add bt slip driver and bugs fix

1. bugs fix
2. add bt h5 slip driver

## Impact

1. bt bridge
2. bt slip features

## Testing

1. local functions test pass

### CODE REVIEW

**Comment by xiaoxiang781216** [comment]
```
====================================================================================
Cmake in present: nrf5340-dk/rpmsghci_nimble_cpuapp,CONFIG_ARM_TOOLCHAIN_GNU_EABI
Configuration/Tool: nrf5340-dk/rpmsghci_nimble_cpuapp,CONFIG_ARM_TOOLCHAIN_GNU_EABI
2024-10-16 11:50:39
------------------------------------------------------------------------------------
  Cleaning...
  Configuring...
  Select HOST_LINUX=y
-- Build type:  
-- Host:    Linux/x86_64
-- Target:  NuttX/arm
-- Machine: arm
-- Vendor: none
-- Host:    Linux/x86_64
-- Target:  NuttX/arm
-- Machine: arm
-- C_FLAGS :  -Wall -Wextra
  Disabling CONFIG_ARM_TOOLCHAIN_GNU_EABI
  Enabling CONFIG_ARM_TOOLCHAIN_GNU_EABI
  Building NuttX...
-- Build type:  
-- Host:    Linux/x86_64
-- Target:  NuttX/arm
-- Machine: arm
-- Vendor: none
-- Host:    Linux/x86_64
-- Target:  NuttX/arm
-- Machine: arm
-- C_FLAGS :  -Wall -Wextra
[1/5] cd /github/workspace/sources/nuttx/build/libs/libc/misc && /usr/local/bin/cmake -E touch /github/workspace/sources/nuttx/libs/libc/misc/lib_utsname.c
[2/5] Linking C executable nuttx
FAILED: nuttx 
: && /tools/ccache/bin/arm-none-eabi-gcc  --specs=nosys.specs   -Wl,--entry=__start -nostdlib -Wl,--gc-sections -Wl,--cref -Wl,-Map=nuttx.map CMakeFiles/nuttx.dir/empty.c.obj -o nuttx  -mtune=cortex-m33  -march=armv8-m.main+dsp  -mfpu=fpv5-sp-d16  -mfloat-abi=hard  -fno-common  -Wall  -Wshadow  -Wundef  -nostdlib  -Os  -fno-strict-aliasing  -fomit-frame-pointer  -fstack-protector-all  -fstack-usage  -mthumb  -Wa,-mthumb  -Wa,-mimplicit-it=always  -ffunction-sections  -fdata-sections  -Wno-attributes  -Wno-unknown-pragmas  -Wstrict-prototypes  -Wno-psabi  -T  flash_app.ld.tmp  -Wl,--start-group  openamp/libmetal/lib/libmetal.a  openamp/open-amp/lib/libopen_amp.a  arch/libarch.a  binfmt/libbinfmt.a  drivers/libdrivers.a  fs/libfs.a  libs/libc/libc.a  mm/libmm.a  net/libnet.a  sched/libsched.a  wireless/libwireless.a  boards/libboard.a  apps/libapps.a  apps/wireless/bluetooth/nimble/libnimble.a  apps/builtin/libapps_builtin.a  apps/system/nsh/libapps_nsh.a  apps/system/nsh/libapps_sh.a  /tools/gcc-arm-none-eabi/bin/../lib/gcc/arm-none-eabi/13.2.1/thumb/v8-m.main+fp/hard/libgcc.a  /tools/gcc-arm-none-eabi/bin/../lib/gcc/arm-none-eabi/13.2.1/../../../../arm-none-eabi/lib/thumb/v8-m.main+fp/hard/libm.a  -Wl,--end-group && :
/tools/gcc-arm-none-eabi/bin/../lib/gcc/arm-none-eabi/13.2.1/../../../../arm-none-eabi/bin/ld: boards/libboard.a(nrf53_bringup.c.obj): in function `nrf53_bringup':
nrf53_bringup.c:(.text.nrf53_bringup+0x68): undefined reference to `bt_driver_register_with_id'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```
please fix the link error.

**Comment by xiaoxiang781216** [comment]
please normalize defconfig:
```
Cmake in present: nrf5340-dk/sdc_nimble_cpunet,CONFIG_ARM_TOOLCHAIN_GNU_EABI
Configuration/Tool: nrf5340-dk/sdc_nimble_cpunet,CONFIG_ARM_TOOLCHAIN_GNU_EABI
2024-10-16 20:31:45
------------------------------------------------------------------------------------
  Cleaning...
  Configuring...
  Select HOST_LINUX=y
  Disabling CONFIG_ARM_TOOLCHAIN_GNU_EABI
  Enabling CONFIG_ARM_TOOLCHAIN_GNU_EABI
  Building NuttX...
HEAD detached at pull/14224/merge
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   boards/arm/nrf53/nrf5340-dk/configs/sdc_nimble_cpunet/defconfig
```

File: arch/sim/src/sim/sim_hcisocket.c [hunk 1]
```diff
@@ -332,35 +305,17 @@ int sim_bthcisock_register(int dev_id)
...
+#ifdef CONFIG_DRIVERS_BLUETOOTH
+  ret = bt_driver_register_with_id(&dev->drv, dev_id);
   if (ret < 0)
     {
+      wlerr("bt_driver_register error: %d\n", ret);
       bthcisock_free(dev);
       return ret;
     }
+#else
+# error "Please select CONFIG_DRIVERS_BLUETOOTH"
```
[the remark this answers is not in the document]
Comment (chengkai15) [reply arch/sim/src/sim/sim_hcisocket.c]: done, like this
#else
#  error "Please select CONFIG_DRIVERS_BLUETOOTH"

**Comment by raiden00pl** [comment]
@xiaoxiang781216 @chengkai15 `nrf52840-dk/sdc_nimble` is broken with this PR merged

**Comment by raiden00pl** [comment]
CONFIG_DRIVERS_BLUETOOTH is not selected for `nrf52840-dk`. Did you verify that `CONFIG_DRIVERS_BLUETOOTH` is selected for ALL affected configurations ?

**Comment by raiden00pl** [comment]
this also breaks ALL user configurations that do not set `CONFIG_DRIVERS_BLUETOOTH`

**Comment by xiaoxiang781216** [comment]
sorry, @chengkai15 please review the affect board and ensure it work as before.

**Comment by raiden00pl** [comment]
this also break configurations that use `CONFIG_UART_BTH4` with `/dev/ttyHCI0` and this should be well documented in PR

**Comment by xiaoxiang781216** [comment]
@raiden00pl could you try patch https://github.com/apache/nuttx/pull/14385, which could fix the incompatibility you mention here.

**Pull request merged**